Project1/Map.h

```diff
@@ -0,0 +1,47 @@
+#pragma once
+#include <vector>
+#include "Unit.h"
+#include "GridCell.h"
+#include "CellPath.h"
+#include "GridCell.h"
+#include <map>
+#include <queue>
+
+//Should this class be a singleton?
+/**
+Map class. Holds a Matrix of GridCells and all of the units. Should be initialized at the start of each chapter.
+*/
+class Map {
+private:
+	struct prev_cost_pair {
```

Comment:
std::map only uses operator< for the keys. If you're using it somewhere else, you'll get the same thing using std::pair<int, GridCell*>